tests/phpunit/unit/MediaWiki/Request/EditRequestSaverTest.php

```diff
@@ -0,0 +1,206 @@
+
+		$secondEntityRevision = new EntityRevision( $firstRequestItem, 1234 );
+
+		$secondEditEntity = $this->createMock( EditEntity::class );
+		$secondEditEntity->method( 'attemptSave' )
+			->willReturn( Status::newGood( [ 'revision' => $secondEntityRevision ] ) );
+
+		$thirdEntityRevision = new EntityRevision( $secondRequestItem, 7891 );
+
+		$thirdEditEntity = $this->createMock( EditEntity::class );
+		$thirdEditEntity->method( 'attemptSave' )
+			->willReturn( Status::newGood( [ 'revision' => $thirdEntityRevision ] ) );
+
+		$editEntityFactory = $this->createMock( MediawikiEditEntityFactory::class );
+		$editEntityFactory
+			->expects( $this->exactly( 3 ) )
+			->method( 'newEditEntity' )
+			->withConsecutive(
+				[ $this->user, $firstRequestOtherItem->getId(), false ],
+				[ $this->user, $firstRequestItem->getId(), false ],
+				[ $this->user, $secondRequestItem->getId(), false ],
+			)
+			->willReturnOnConsecutiveCalls(
+				$firstEditEntity,
+				$secondEditEntity,
+				$thirdEditEntity
+			);
+
+		$saver = $this->newRequestSaver( $editEntityFactory, $itemReconciler );
+
+		// gets updated once saved
+		$this->assertFalse( $otherItem->getRevision() );
+
+		$status = $saver->persistEdits( [
+			$editRequestOne,
+			$editRequestTwo
+		], 'token', $this->user );
+
+		$this->assertTrue( $status->isGood() );
+		$this->assertEquals( $secondEntityRevision, $status->getValue()[0]['revision'] );
+		$this->assertEquals( $thirdEntityRevision, $status->getValue()[1]['revision'] );
+
+		// updated from save
+		$this->assertEquals( $firstEntityRevision->getRevisionId(), $otherItem->getRevision() );
+	}
+
+	public function testEditRequestSavesWith_OtherItemsThatFail(): void {
+		$firstRequestItem = FluidItem::withId( 'Q1' )->item();
+		$firstRequestOtherItem = FluidItem::withId( 'Q2' )->item();
+
+		$otherItem = new ReconciliationServiceItem( $firstRequestOtherItem, false );
+		$editRequestOne = new EditRequest( $this->reconcileProperty, $firstRequestItem, [ $otherItem ] );
+
+		$itemReconciler = $this->createMock( ItemReconciler::class );
+
+		$firstEditEntity = $this->createMock( EditEntity::class );
+		$firstEditEntity->method( 'attemptSave' )
+			->willReturn( Status::newFatal( 'not cool' ) );
+
+		$editEntityFactory = $this->createMock( MediawikiEditEntityFactory::class );
+		$editEntityFactory
+			->expects( $this->exactly( 1 ) )
+			->method( 'newEditEntity' )
+			->with( $this->user, $firstRequestOtherItem->getId(), false )
+			->willReturn( $firstEditEntity );
+
+		$saver = $this->newRequestSaver( $editEntityFactory, $itemReconciler );
+
+		// gets updated once saved
+		$this->assertFalse( $otherItem->getRevision() );
+
+		$status = $saver->persistEdits( [
+			$editRequestOne,
+		], 'token', $this->user );
+
+		$this->assertFalse( $status->isGood() );
+		$this->assertFalse( $status->isOK() );
+
+		$this->assertEquals( 'not cool', $status->getErrors()[0]['message'] );
+		$this->assertEquals( 'error', $status->getErrors()[0]['type'] );
```

Comment:
Could also be `assertSame()`, I think, though for these strings it shouldn’t make a difference IIUC.